modules/apps/address/address-impl/src/main/java/com/liferay/address/internal/util/CompanyCountriesUtil.java

```diff
@@ -224,6 +229,35 @@ public static void processCountryRegions(
 		}
 	}
 
+	public static void updateRegionCounter(Connection connection)
+		throws Exception {
+
+		try (Statement statement = connection.createStatement();
+			ResultSet resultSet = statement.executeQuery(
+				"select max(regionId) from Region")) {
+
+			if (resultSet.next()) {
+				CounterLocalServiceUtil.increment(
```

Comment:
This call is incorrect. When calling
`public static long increment(String name, int size) {`

The parameter size is the increment to do over the current counter value.

So passing the max id as size is not right. We must pass the difference between the current max id and the counter id. You have several examples in upgrades like:
https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeListTypeCompanyId.java#L107-L132

Using this logic you will also update the counter when really needed, depending on the customer's original version, it can be not needed.